### PR TITLE
Run tests with specific aqa-tests releases

### PR DESCRIPTION
**Pull request by sophia-guo**
Run tests with specific aqa-tests releases, which is consistent with openjdk releases.

Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

This allows us to pick a particular AQA release branch/tag and also set a particular test environment properties

### CODE REVIEW

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -288,9 +288,10 @@ class Build {
                                     context.string(name: 'UPSTREAM_JOB_NUMBER', value: "${env.BUILD_NUMBER}"),
                                     context.string(name: 'UPSTREAM_JOB_NAME', value: "${env.JOB_NAME}"),
                                     context.string(name: 'JDK_VERSION', value: "${jobParams.JDK_VERSIONS}"),
-                                    context.string(name: 'RELEASE_TAG', value: "${buildConfig.SCM_REF}"),
                                     context.string(name: 'LABEL_ADDITION', value: additionalTestLabel),
                                     context.string(name: 'KEEP_REPORTDIR', value: "${buildConfig.KEEP_TEST_REPORTDIR}"),
+                                    context.string(name: 'ADOPTOPENJDK_BRANCH', value: "${buildConfig.KEEP_TEST_REPORTDIR}"),
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:293]: The default values for both of these seem incorrect

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 2]
```diff
@@ -315,6 +316,12 @@ class Build {
         List dynamicList = buildConfig.DYNAMIC_LIST
         List numMachines = buildConfig.NUM_MACHINES
         def enableTestDynamicParallel = Boolean.valueOf(buildConfig.ENABLE_TESTDYNAMICPARALLEL)
+        def aqa_branch = "master"
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:319]: naming convention aqaBranch?

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 3]
```diff
@@ -315,6 +316,12 @@ class Build {
...
+        def useTestEnv = false
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:320]: So this is saying that if the user passes in the SCM ref it's automatically also the aqaBranch and we say we're using a testEnv?

I'm not 100% sure on the logic...

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 4]
```diff
@@ -363,14 +370,14 @@ class Build {
...
-                                            context.string(name: 'RELEASE_TAG', value: "${buildConfig.SCM_REF}"),
                                             context.string(name: 'JDK_REPO', value: jdkRepo),
                                             context.string(name: 'JDK_BRANCH', value: jdkBranch),
                                             context.string(name: 'OPENJ9_BRANCH', value: openj9Branch),
                                             context.string(name: 'LABEL_ADDITION', value: additionalTestLabel),
                                             context.string(name: 'KEEP_REPORTDIR', value: "${keep_test_reportdir}"),
                                             context.string(name: 'PARALLEL', value: parallel),
-                                            context.string(name: 'NUM_MACHINES', value: "${numMachinesPerTest}"),
+                                            context.string(name: 'USE_TESTENV_PROPERTIES', value: useTestEnv),
+                                            context.string(name: 'ADOPTOPENJDK_BRANCH', value: aqa_branch),
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:380]: Is that the right key/label?  should be AQA_BRANCH?

File: pipelines/build/common/weekly_release_pipeline.groovy [hunk 5]
```diff
@@ -43,6 +43,7 @@ stage("Submit Release Pipelines") {
                       parameters: [
                           string(name: 'releaseType',        value: 'Release'),
                           string(name: 'scmReference',       value: scmRef),
+                          string(name: 'aqa-testsReference',   value: scmRef),
```
Comment (karianna) [inline pipelines/build/common/weekly_release_pipeline.groovy:46]: aqaTestsReference (same comment for other occurrences as well)

**Review by karianna — CHANGES_REQUESTED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 6]
```diff
@@ -315,6 +314,12 @@ class Build {
         List dynamicList = buildConfig.DYNAMIC_LIST
         List numMachines = buildConfig.NUM_MACHINES
         def enableTestDynamicParallel = Boolean.valueOf(buildConfig.ENABLE_TESTDYNAMICPARALLEL)
+        def aqaBranch = "master"
+        def useTestEnv = false
+        if (buildConfig.SCM_REF && && buildConfig.AQA_REF) {
+            aqaBranch = buildConfig.AQA_REF
+            useTestEnv = true
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:321]:
```suggestion
            useTestEnvProperties = true
```

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 7]
```diff
@@ -315,6 +314,12 @@ class Build {
         List dynamicList = buildConfig.DYNAMIC_LIST
         List numMachines = buildConfig.NUM_MACHINES
         def enableTestDynamicParallel = Boolean.valueOf(buildConfig.ENABLE_TESTDYNAMICPARALLEL)
+        def aqaBranch = "master"
+        def useTestEnv = false
+        if (buildConfig.SCM_REF && && buildConfig.AQA_REF) {
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:319]:
```suggestion
        if (buildConfig.SCM_REF && buildConfig.AQA_REF) {
```

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 8]
```diff
@@ -363,14 +368,14 @@ class Build {
...
                                             context.string(name: 'UPSTREAM_JOB_NAME', value: "${env.JOB_NAME}"),
-                                            context.string(name: 'RELEASE_TAG', value: "${buildConfig.SCM_REF}"),
                                             context.string(name: 'JDK_REPO', value: jdkRepo),
                                             context.string(name: 'JDK_BRANCH', value: jdkBranch),
                                             context.string(name: 'OPENJ9_BRANCH', value: openj9Branch),
                                             context.string(name: 'LABEL_ADDITION', value: additionalTestLabel),
                                             context.string(name: 'KEEP_REPORTDIR', value: "${keep_test_reportdir}"),
                                             context.string(name: 'PARALLEL', value: parallel),
-                                            context.string(name: 'NUM_MACHINES', value: "${numMachinesPerTest}"),
+                                            context.string(name: 'USE_TESTENV_PROPERTIES', value: useTestEnv),
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:377]:
```suggestion
                                            context.string(name: 'USE_TESTENV_PROPERTIES', value: useTestEnvProperties),
```

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 9]
```diff
@@ -363,14 +368,14 @@ class Build {
...
+                                            context.string(name: 'ADOPTOPENJDK_BRANCH', value: aqaBranch),
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:378]:
```suggestion
                                            // We keep the ADOPTOPENJDK label for historical compatibility.
                                            context.string(name: 'ADOPTOPENJDK_BRANCH', value: aqaBranch),
```

**Review by karianna — CHANGES_REQUESTED** [review]

File: pipelines/build/common/weekly_release_pipeline.groovy (hunk 5 above)
Comment (sophia-guo) [reply pipelines/build/common/weekly_release_pipeline.groovy:46]: Removed. The change is not necessary.

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 9 above)
Comment (sophia-guo) [reply pipelines/build/common/openjdk_build_pipeline.groovy:379]: Add the comment to the definition of aqaBranch https://github.com/adoptium/ci-jenkins-pipelines/pull/229/files#diff-b4bb07e2f448f9ad2986d69423ad0b704dcdc4d217d947f465e6ace3fde7a4b3R320

**Comment by andrew-m-leonard** [comment]
run tests

File: pipelines/build/common/create_job_from_template.groovy [hunk 10]
```diff
@@ -93,7 +93,8 @@ pipelineJob("$buildFolder/$JOB_NAME") {
                 <dt><strong>TEST_LIST</strong></dt><dd>Comma separated list of tests, i.e: sanity.openjdk,sanity.perf,sanity.system</dd>
                 <dt><strong>DYNAMIC_LIST</strong></dt><dd>Comma separated list of tests, i.e: sanity.openjdk,sanity.perf,sanity.system</dd>
                 <dt><strong>NUM_MACHINES</strong></dt><dd>The number of machines for parallel=dynamic</dd>
-                <dt><strong>SCM_REF</strong></dt><dd>Source code ref to build, i.e branch, tag, commit id</dd>
+                <dt><strong>SCM_REF</strong></dt><dd>Source code ref to build, i.e branch, tag, commit id.</dd>
+                <dt><strong>AQA_REF</strong></dt><dd>Specific aqa-tests releases or branch.</dd>
```
Comment (smlambert) [inline pipelines/build/common/create_job_from_template.groovy:97]:
```suggestion
                <dt><strong>AQA_REF</strong></dt><dd>Specific aqa-tests release or branch.</dd>
```

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 11]
```diff
@@ -315,6 +314,13 @@ class Build {
         List dynamicList = buildConfig.DYNAMIC_LIST
         List numMachines = buildConfig.NUM_MACHINES
         def enableTestDynamicParallel = Boolean.valueOf(buildConfig.ENABLE_TESTDYNAMICPARALLEL)
+        def aqaBranch = "master"
+        def useTestEnvProperties = false
+        if (buildConfig.SCM_REF && buildConfig.AQA_REF) {
+            // We keep the ADOPTOPENJDK label for historical compatibility.
```
Comment (smlambert) [inline pipelines/build/common/openjdk_build_pipeline.groovy:320]: do not think this suggested comment is needed anymore here

**Review by smlambert — APPROVED** [review]
aside from a couple of minor points, LGTM

**Comment by sophia-guo** [comment]
@karianna updated and ready for review.

**Review by karianna — APPROVED** [review]
Really clean patch, love it!

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 11 above)
Comment (karianna) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: Correct, can be removed.

**Pull request merged**